lib/Test/APIcast.pm

```diff
@@ -66,7 +83,7 @@ sub get_random_port {
     if (!defined $ServerPort) {
         bail_out "Cannot find an available listening port number after $tries attempts.\n";
     }
-
+    file_unlock();
```

Comment:
This won't be called if the script does `bail_out` before. I'd try really hard to not use locks unless absolutely necessary. And if so then use some library that does it already.